SQL/paradise_schema.sql

```diff
@@ -367,17 +367,22 @@ DROP TABLE IF EXISTS `library`;
 /*!40101 SET @saved_cs_client     = @@character_set_client */;
 /*!40101 SET character_set_client = utf8 */;
 CREATE TABLE `library` (
-  `id` int(11) NOT NULL AUTO_INCREMENT,
-  `author` mediumtext COLLATE utf8mb4_unicode_ci NOT NULL,
-  `title` mediumtext COLLATE utf8mb4_unicode_ci NOT NULL,
-  `content` mediumtext COLLATE utf8mb4_unicode_ci NOT NULL,
-  `category` mediumtext COLLATE utf8mb4_unicode_ci NOT NULL,
-  `ckey` varchar(32) COLLATE utf8mb4_unicode_ci NOT NULL,
-  `flagged` int(11) NOT NULL,
-  PRIMARY KEY (`id`),
-  KEY `ckey` (`ckey`),
-  KEY `flagged` (`flagged`)
-) ENGINE=InnoDB AUTO_INCREMENT=4537 DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+	`id` INT(11) NOT NULL AUTO_INCREMENT,
+	`author` MEDIUMTEXT NOT NULL COLLATE 'utf8mb4_unicode_ci',
+	`title` MEDIUMTEXT NOT NULL COLLATE 'utf8mb4_unicode_ci',
+	`content` MEDIUMTEXT NOT NULL COLLATE 'utf8mb4_unicode_ci',
+	`ckey` VARCHAR(32) NULL DEFAULT '' COLLATE 'utf8mb4_unicode_ci',
+	`reports` MEDIUMTEXT NOT NULL COLLATE 'utf8mb3_general_ci',
+	`summary` MEDIUMTEXT NOT NULL COLLATE 'utf8mb3_general_ci',
+	`ratings` DOUBLE NULL DEFAULT '0',
```

Comment:
So is this a rating number (sorta like post score), or is this the total number of rates? If the latter, is it worth just running `length()` on raters?

Reply:
I should probably modify it to just be `rating` tbh. It is the avg of all the ratings, the intention is to significantly reduce the calculations needed in order to display ratings in the menu. We only have to calculate the rating once when someone makes a new rating instead of doing it 25 times per library page.